snooty/diagnostics.py

```diff
@@ -851,6 +851,22 @@ def __init__(
         )
 
 
+class CardIconString(Diagnostic):
+    severity = Diagnostic.Level.warning
+
+    def __init__(
+        self,
+        image_argument: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            f"""Card icons are migrating to strings instead of path names. If {image_argument} can be a string, use that instead.""",
```

Comment:
(Non-blocking) Would it be worth including more information on what the "string" is supposed to be or how to know what the string is supposed to be?

Also, I know that we have a strategy for handling marketing's icons through the web assets URL, but do we have a strategy for other icons like for drivers/languages/[integrations](https://www.mongodb.com/docs/languages/java/#integrations--community-maintained-)? If we don't currently support all existing icons by using strings, I'm afraid of causing noise and confusion through this warning for card icons that can't immediately be updated. Should we consider adding this warning at a later time once we fully support everything?

Reply:
that sounds good!

Reply:
i am wondering if the warning that a dark mode icon hasn't been included should be silenced until we are closer to releasing dark mode?

Reply:
I think I'm okay with waiting if we need the time to sort out all of the dark mode icons for cards

Reply:
Removed the warning!